pure_pursuit_wrapper/src/pure_pursuit_wrapper_node.cpp

```diff
@@ -28,14 +28,26 @@ int main(int argc, char** argv) {
   ros::NodeHandle nh("");
 
   pure_pursuit_wrapper::PurePursuitWrapper PurePursuitWrapper(nh);
+  ros::CARMANodeHandle::setSpinRate(10);
 
   // Approximate time of 100ms used because NDT outputs at 10Hz
   message_filters::Synchronizer<pure_pursuit_wrapper::PurePursuitWrapper::SyncPolicy> sync(pure_pursuit_wrapper::PurePursuitWrapper::SyncPolicy(100), PurePursuitWrapper.pose_sub, PurePursuitWrapper.trajectory_plan_sub);
   sync.registerCallback(boost::bind(&pure_pursuit_wrapper::PurePursuitWrapper::TrajectoryPlanPoseHandler, &PurePursuitWrapper, _1, _2));
 
-  while (ros::ok() && !PurePursuitWrapper.shutting_down_) {
-    ros::spinOnce();
-  }
+  //ros::Rate r(10);
+  //while (ros::ok() && !PurePursuitWrapper.shutting_down_) {
+  //  PurePursuitWrapper.PublishPluginDiscovery();
+  //  ros::spinOnce();
+  //  r.sleep();
+ // }
+
+  ros::CARMANodeHandle::setSpinCallback([this]() -> bool
+  {
+  if (!PurePursuitWrapper.shutting_down_)
+  {PurePursuitWrapper.PublishPluginDiscovery();}
```

Comment:
If you're capturing `[this]` in the lambda, I think you can just directly reference `shutting_down_` and `PublishPluginDiscovery()`  (or at worst: `this.shutting_down_` or `this.PublishPluginDiscovery()` rather than referring to them by class like this. This seems like these are static members rather than members of `this`.